src/SNLS_device_forall.h

```diff
@@ -50,14 +56,97 @@ end,                                             \
 
 namespace snls {
 
-   typedef RAJA::View<bool, RAJA::Layout<1> > rview1b;
-   typedef RAJA::View<double, RAJA::Layout<1> > rview1d;
-   typedef RAJA::View<double, RAJA::Layout<2> > rview2d;
-   typedef RAJA::View<double, RAJA::Layout<3> > rview3d;
-   typedef RAJA::View<const bool, RAJA::Layout<1> > crview1b;
-   typedef RAJA::View<const double, RAJA::Layout<1> > crview1d;
-   typedef RAJA::View<const double, RAJA::Layout<2> > crview2d;
-   typedef RAJA::View<const double, RAJA::Layout<3> > crview3d;
+   // Provide some simple shortcuts in-case people need something beyond the default
+   template<typename T>
+   using rview1 = RAJA::View<T, RAJA::Layout<1>>;
+   template<typename T>
+   using rview2 = RAJA::View<T, RAJA::Layout<2>>;
+   template<typename T>
+   using rview3 = RAJA::View<T, RAJA::Layout<3>>;
+
+   using rview1b = rview1<bool>;
+   using rview1d = rview1<double>;
+   using rview2d = rview2<double>;
+   using rview3d = rview3<double>;
+   using crview1d = rview1<const double>;
```

Comment:
Maybe we should think of another name for these aliases (or just leave them out)? I lost count of the number of times I thought "rview1d" was aliased to a one-dimensional RAJA view.

Reply:
Would changing it to something like `rview1f64` be clearer?  I just want something to avoid having to have such long names due to all the templates... Although, it's not as bad now with the templated aliased type...

Reply:
Yeah I think that would be better.  I also don't really mind just removing those aliases, since they don't really save much space.